components/eventing-controller/controllers/subscription/nats/reconciler.go

```diff
@@ -251,17 +226,49 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Resu
 	return ctrl.Result{}, nil
 }
 
+// handleSubscriptionDeletion deletes the subscription and removes the finalizer
+func (r *Reconciler) handleSubscriptionDeletion(ctx context.Context, subscription *eventingv1alpha1.Subscription, log *zap.SugaredLogger) error {
+	if utils.ContainsString(subscription.ObjectMeta.Finalizers, Finalizer) {
+		if err := r.Backend.DeleteSubscription(subscription); err != nil {
+			log.Errorw("delete subscription failed", "error", err)
```

Comment:
Let's make this clearer: `subscription` means `Kyma eventing` subscription, `NATS subscription` means NATS subscription.

```suggestion
			log.Errorw("delete NATS subscription failed", "error", err)
```

Reply:
done